File: 03 Divide and Conquer/Maximum Contiguous Sum/ex01e2.cpp
#include<iostream>
using namespace std;

/**
 * @brief Find maximum contiguous sum by using Kadane's Algorithm */
int main()
{
    // This makes std::cin and std::cout faster
    ios_base::sync_with_stdio(false); cin.tie(0);
    // Initialize value
    int maxSum = -2147483648; // Lowest int value
    int currentSum = 0;
    // Input each number and using Kadane's Algorithm
    int N;
    cin >> N;
    while(N--)
    {
        int num;
        cin >> num;
        // Find currentSum and maxSum
        currentSum += num;
        maxSum = max(currentSum, maxSum);
        // If currentSum is negative, resets currentSum to 0
        currentSum = max(0, currentSum);
    }
    // Output
    cout << maxSum;
    return 0;
}

File: 03 Divide and Conquer/Maximum Contiguous Sum/ex01e2.py
import sys
maxSum = -1e6
currentSum = 0
n = int(sys.stdin.readline())
for num in sys.stdin.readline().split():
    currentSum += int(num)
    maxSum = max(currentSum, maxSum)
    currentSum = max(0, currentSum)
sys.stdout.write(str(maxSum))